File: BEmu3/MarketDataRequest/ElementMarketNull.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace BEmu3.MarketDataRequest
{
    public class ElementMarketNull : Element
    {
        private readonly string _name;

        internal ElementMarketNull(string name)
        {
            this._name = name;
        }

        public override Name Name { get { return new Name(this._name); } }
        public override int NumElements { get { return 0; } }
        public override int NumValues { get { return 0; } }
        public override bool IsArray { get { return true; } }
        public override bool IsComplexType { get { return false; } }
        public override bool IsNull { get { return true; } }

        internal override StringBuilder PrettyPrint(int tabIndent)
        {
            return new StringBuilder();
        }

    }
}

File: BEmu_cpp/src/MarketDataRequest/ElementMarketNull.cpp
//------------------------------------------------------------------------------
// <copyright project="BEmu_cpp" file="src/MarketDataRequest/ElementMarketNull.cpp" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

#include "MarketDataRequest/ElementMarketNull.h"

namespace BEmu
{
	namespace MarketDataRequest
	{
		ElementMarketNull::ElementMarketNull(std::string name)
		{
			this->_name = name;
		}

		Name ElementMarketNull::name() const
		{
			return Name(this->_name.c_str());
		}

		size_t ElementMarketNull::numValues() const
		{
			return 0;
		}

		size_t ElementMarketNull::numElements() const
		{
			return 0;
		}

		bool ElementMarketNull::isNull() const
		{
			return true;
		}

		bool ElementMarketNull::isArray() const
		{
			return true;
		}

		bool ElementMarketNull::isComplexType() const
		{
			return false;
		}

		std::ostream& ElementMarketNull::print(std::ostream& stream, int level, int spacesPerLevel) const
		{
			return stream;
		}

	}
}